lib/ex_unit/lib/ex_unit/server.ex

```diff
@@ -68,6 +83,39 @@ defmodule ExUnit.Server do
       %{config | captured_devices: Set.delete(config.captured_devices, device)}}
   end
 
+  def handle_call({:log_capture_on, pid, args}, _from, config) do
+    ref = Process.monitor(pid)
+    try do
+      GenEvent.add_mon_handler(Logger, {Console, ref}, args)
+    catch
+      :exit, :noproc ->
+        Process.demonitor(ref, [:flush])
+        {:reply, {:error, :no_logger}, config}
+    else
+      :ok ->
+        case Map.put(config.log_captures, ref, :ok) do
+          refs when map_size(refs) > 1 ->
+            {:reply, {:ok, ref}, %{config | log_captures: refs}}
+          refs ->
+            status = Logger.remove_backend(:console)
+            {:reply, {:ok, ref}, %{config | log_captures: refs, log_status: status}}
+        end
+    end
+  end
+
+  def handle_call({:log_capture_off, ref}, _from, config) do
+    Process.demonitor(ref, [:flush])
+    case Map.pop(config.log_captures, ref) do
+      {:ok, refs} ->
+        maybe_add_console(refs, config.log_status)
+        GenEvent.remove_handler(Logger, {Console, ref}, nil)
```

Comment:
The reason may be `{:error, :not_found}` here. We should check the return value and flush the `:gen_event_EXIT` message (otherwise its possible to to receive it later and store it forever never to replace the console).